frontend/src/shared/util/rich-text-utils.tsx

```diff
@@ -1,177 +1,220 @@
+            return;
+        }
 
-                for (let x = 0; x < options.length; x++) {
-                    const item = options[x];
+        // update link
+        editor
+            ?.chain()
+            .focus()
+            .extendMarkRange('link')
+            .setLink({ href: url })
+            .run();
+    }, [editor]);
 
-                    // Read the css 'content' values and generate aria labels
-                    item.setAttribute('aria-label', item['dataset'].value);
-                    item.addEventListener('keyup', (e) => {
-                        if (e.keyCode === 13) {
-                            item.click();
-                            optionsContainer.setAttribute('aria-hidden', 'true');
-                        }
-                    });
-                }
+    if (!editor) {
+        return null;
+    }
 
-                label.addEventListener('keyup', (e) => {
-                    if (e.keyCode === 13) {
-                        label.click();
-                        optionsContainer.setAttribute('aria-hidden', 'false');
-                    }
-                });
-            }
-        });
-    }, []);
+    editor.on('update', ({ editor }) => {
+        props.onChange(editor.getHTML());
+    });
 
     return (
-        <ReactQuill
-            modules={modules}
-            formats={formats}
-            value={props.value}
-            onChange={props.onChange}
-        />
+        <Container>
+            <ColumnLayout columns={1} borders='horizontal'>
+                <SpaceBetween direction='horizontal' size={'s'}>
+                    <Button
+                        onClick={() =>
+                            editor.chain().focus().toggleHeading({ level: 1 }).run()
+                        }
+                        ariaLabel={'Toggle heading level 1'}
+                        variant='inline-link'
+                    >
+                        <Icon
+                            variant={
+                                editor.isActive('heading', { level: 1 }) ? 'normal' : 'subtle'
+                            }
+                            svg={
+                                <svg viewBox='0 0 18 18'>
+                                    <title>Heading level 1 icon</title>
+                                    {
+                                        // eslint-disable-next-line spellcheck/spell-checker
+                                    }<path d='M10,4V14a1,1,0,0,1-2,0V10H3v4a1,1,0,0,1-2,0V4A1,1,0,0,1,3,4V8H8V4a1,1,0,0,1,2,0Zm6.06787,9.209H14.98975V7.59863a.54085.54085,0,0,0-.605-.60547h-.62744a1.01119,1.01119,0,0,0-.748.29688L11.645,8.56641a.5435.5435,0,0,0-.022.8584l.28613.30762a.53861.53861,0,0,0,.84717.0332l.09912-.08789a1.2137,1.2137,0,0,0,.2417-.35254h.02246s-.01123.30859-.01123.60547V13.209H12.041a.54085.54085,0,0,0-.605.60547v.43945a.54085.54085,0,0,0,.605.60547h4.02686a.54085.54085,0,0,0,.605-.60547v-.43945A.54085.54085,0,0,0,16.06787,13.209Z'></path>
```

Comment:
it disables eslint on that specific line. for some reason eslint is detecting spelling mistakes on that line but that line is just an SVG for the icons. rather than adding random characters to the eslint dictionary, I opted to disable it here.

Reply:
How was this SVG generated? If someone else needed to create an SVG for a new icon what would that entail?

Reply:
I directly took it from the SVG that was generated from the HTML of the previous library. Copy-paste. To generate future icons, you would have to either copy paste them from somewhere or use an Icon library.